### PR TITLE
Junit tests

### PR DESCRIPTION
**Pull request by greatpanda1603**

### CODE REVIEW

File: core/src/test/java/greencity/service/impl/FavoritePlaceServiceImplTest.java [hunk 1]
```diff
@@ -37,6 +38,8 @@
     @Mock
     private FavoritePlaceDtoMapper favoritePlaceDtoMapper;
     @Mock
+    private FavoritePlaceMapper favoritePlaceMapper;
```
Comment (zendermean) [inline core/src/test/java/greencity/service/impl/FavoritePlaceServiceImplTest.java:41]: All these tests use junit4. You need to refactor all of them on junit5.

1) All core annotations are located in the **org.junit.jupiter.api** package in the junit-jupiter-api module.
2) All JUnit Jupiter assertions are static methods in the **org.junit.jupiter.api.Assertions** class.

In JUnit 5, to test **exception** cases (instead @Test(expected = ...class) you should use **org.junit.jupiter.api.Assertions.assertThrows()** method.
simple example:
@Test
void testExpectedException() {
  Assertions.assertThrows(NumberFormatException.class, () -> {
    Integer.parseInt("seven");
  });
}

File: core/src/test/java/greencity/service/impl/GoalServiceImplTest.java [hunk 2]
```diff
@@ -1,42 +1,49 @@
...
+import greencity.ModelUtils;
+import greencity.constant.AppConstant;
+import greencity.dto.user.UserGoalResponseDto;
+import greencity.entity.UserGoal;
+import greencity.mapping.UserGoalResponseDtoMapper;
+import greencity.repository.CustomGoalRepo;
+import greencity.repository.GoalRepo;
+import greencity.service.LanguageService;
+import java.util.Optional;
 import static org.junit.Assert.assertEquals;
```
Comment (zendermean) [inline core/src/test/java/greencity/service/impl/GoalServiceImplTest.java:12]: The same thing.

File: core/src/test/java/greencity/mapping/EcoNewsAuthorDtoMapperTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package greencity.mapping;
+
+import greencity.ModelUtils;
+import greencity.dto.user.EcoNewsAuthorDto;
+import greencity.entity.User;
+import static org.junit.Assert.assertEquals;
```
[the remark this answers is not in the document]
Comment (greatpanda1603) [reply core/src/test/java/greencity/mapping/EcoNewsAuthorDtoMapperTest.java]: fixed

File: core/src/main/java/greencity/dto/user/UserGoalResponseDto.java [hunk 4]
```diff
@@ -2,12 +2,14 @@
 
 import greencity.entity.enums.GoalStatus;
 import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Data;
```
[the remark this answers is not in the document]
Comment (greatpanda1603) [reply core/src/main/java/greencity/dto/user/UserGoalResponseDto.java]: @Data annotation was replaced by @Getter / @Setter and @EqualsAndHashCode (if needed) in all entities and DTOs used in mappers, other items will be changed within the scope of further refactoring

File: core/src/main/java/greencity/entity/UserGoal.java [hunk 5]
```diff
@@ -4,19 +4,16 @@
...
 import javax.persistence.*;
-import lombok.AllArgsConstructor;
-import lombok.Data;
-import lombok.NoArgsConstructor;
-import lombok.ToString;
+import lombok.*;
 import org.springframework.format.annotation.DateTimeFormat;
 
 @Entity
 @Data
```
[the remark this answers is not in the document]
Comment (greatpanda1603) [reply core/src/main/java/greencity/entity/UserGoal.java:11]: fixed

File: core/src/test/java/greencity/mapping/FavoritePlaceMapperTest.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+import greencity.ModelUtils;
+import greencity.dto.favoriteplace.FavoritePlaceDto;
+import greencity.entity.FavoritePlace;
+import greencity.entity.Place;
+import greencity.entity.User;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import static org.junit.Assert.assertEquals;
```
[the remark this answers is not in the document]
Comment (greatpanda1603) [reply core/src/test/java/greencity/mapping/FavoritePlaceMapperTest.java]: fixed

File: core/src/test/java/greencity/mapping/GoalDtoMapperTest.java [hunk 7]
```diff
@@ -1,32 +1,24 @@
 package greencity.mapping;
 
+import greencity.ModelUtils;
+import greencity.entity.localization.GoalTranslation;
 import static org.junit.Assert.assertEquals;
```
[the remark this answers is not in the document]
Comment (greatpanda1603) [reply core/src/test/java/greencity/mapping/GoalDtoMapperTest.java]: fixed

File: core/src/test/java/greencity/service/impl/GoalServiceImplTest.java (hunk 2 above)
Comment (greatpanda1603) [reply core/src/test/java/greencity/service/impl/GoalServiceImplTest.java:12]: fixed

File: core/src/test/java/greencity/service/impl/FavoritePlaceServiceImplTest.java (hunk 1 above)
Comment (greatpanda1603) [reply core/src/test/java/greencity/service/impl/FavoritePlaceServiceImplTest.java:41]: In the scope of current task there was aim to refactor mappers (remove services and make them follow the unified scheme - extending the AbstractConverter) and cover them with respective tests. Migration of all ServiceImpl Tests to JUnit 5 might be added as a task to the backlog to be solved if the next sprint.

File: core/src/main/java/greencity/dto/advice/AdviceDTO.java [hunk 8]
```diff
@@ -4,14 +4,14 @@
 import greencity.dto.user.HabitDictionaryDto;
 import javax.validation.constraints.Min;
 import javax.validation.constraints.NotBlank;
-import lombok.AllArgsConstructor;
-import lombok.Data;
-import lombok.NoArgsConstructor;
+import lombok.*;
```
Comment (vterlyha) [inline core/src/main/java/greencity/dto/advice/AdviceDTO.java:7]: Better avoid this (*). You can configure Intellij IDEA not to group imports together

File: core/src/main/java/greencity/service/impl/FavoritePlaceServiceImpl.java [hunk 9]
```diff
@@ -30,6 +31,7 @@
     private UserService userService;
     private PlaceService placeService;
     FavoritePlaceDtoMapper favoritePlaceDtoMapper;
+    FavoritePlaceMapper favoritePlaceMapper;
```
Comment (vterlyha) [inline core/src/main/java/greencity/service/impl/FavoritePlaceServiceImpl.java:34]: You need to inject only ModelMapper - not several mapper and use it as modelMapper.map(). And based on the classes you are passing - modelmapper should decide what converter to use.

File: core/src/main/java/greencity/service/impl/GoalServiceImpl.java [hunk 10]
```diff
@@ -34,4 +56,22 @@ public GoalServiceImpl(GoalTranslationRepo goalTranslationRepo, ModelMapper mode
...
+            Goal goal = goalRepo
+                .findById(userGoal
+                    .getGoal().getId()).orElseThrow(() -> new GoalNotFoundException(GOAL_NOT_FOUND_BY_ID));
+            userGoalResponseDto.setText(goalTranslationRepo.findByGoalAndLanguageCode(goal, languageCode)
+                .orElseThrow(() -> new GoalNotFoundException(GOAL_NOT_FOUND_BY_LANGUAGE_CODE)).getText());
+        } else if (userGoal.getGoal() == null) {
+            CustomGoal customGoal = customGoalRepo.findById(userGoal
+                .getCustomGoal().getId()).orElseThrow(() -> new NotFoundException(CUSTOM_GOAL_NOT_FOUND_BY_ID));
+            userGoalResponseDto.setText(customGoal.getText());
+        }
```
Comment (vterlyha) [inline core/src/main/java/greencity/service/impl/GoalServiceImpl.java:74]: Please, move this to a separate private method

File: core/src/main/java/greencity/dto/advice/AdviceDTO.java (hunk 8 above)
Comment (greatpanda1603) [reply core/src/main/java/greencity/dto/advice/AdviceDTO.java]: fixed, made necessary configurations Intellij IDEA

File: core/src/main/java/greencity/service/impl/FavoritePlaceServiceImpl.java (hunk 9 above)
Comment (greatpanda1603) [reply core/src/main/java/greencity/service/impl/FavoritePlaceServiceImpl.java:34]: In order to follow the unified approach currently custom mappers extend AbstractConverter and need to override convert method to be further used. All of them were added to MapperConfig class.

File: core/src/main/java/greencity/service/impl/GoalServiceImpl.java (hunk 10 above)
Comment (greatpanda1603) [reply core/src/main/java/greencity/service/impl/GoalServiceImpl.java]: refactored

File: core/src/main/java/greencity/service/impl/FavoritePlaceServiceImpl.java (hunk 9 above)
Comment (greatpanda1603) [reply core/src/main/java/greencity/service/impl/FavoritePlaceServiceImpl.java]: Redundant mappers were removed, service class and respective test were refactored using modelMapper.map().

**Pull request merged**